.github/workflows/_test_te.yaml

```diff
@@ -106,6 +168,7 @@ jobs:
         id: submit
         shell: bash -O expand_aliases -x -e {0}
         run: |
+          cd $GITHUB_WORKSPACE
```

Comment:
Why do you `cd` to the workspace here, and in several other places?

Reply:
in _ci.yaml file since we run multiple build and test jobs for jax, t5x, pax I did notice that sometimes the current working dir gets changed because of previous job and is not the GITHUB_WORKSPACE dir due to which it either throws error for repo files not found (e.g. to_json.sh) or the files get dumped at different locations and when we are saving the artifact it either complains about the folder/file doesn't exist or the dumped file is missing. for eg. in this run: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/6741297375/job/18326079678#step:9:45

Reply:
Interesting. I never noticed this before, have you seen this before @yhtang ?

I'm surprised since it's not obvious previous jobs change the working directory. Out of curiosity, does setting the [default for the entire job work](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-setting-default-run-step-options-for-a-job)?

Reply:
Each GitHub Actions "job" runs in a clean VM so it's unlikely that a previous job can chang the working path of a later job. However, an earlier job *step* can change that for a later step.

For the particular job that failed, it looks like the reason is that the repo is not checked out at all. Note that GitHub Actions does not by default check out the triggering repo. It needs to be done using `actions/checkout`.

Reply:
Yeah I did add checkout step too but sometimes the current working dir gets changed maybe because of earlier job step. I do cd just to make sure I am dumping the data in the right dir to save it into artifact, it's just a safe thing to do without causing any issue.

Reply:
That's perfectly fine.